### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by jr0me**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-agent/issues/107|

## Description

To avoid confusion this PR modifies the `README.md` to mention that this is not the repository for the released Wazuh-Agent.

We also add more info for compiling the project (missing system packages).

I would still recommended to move ongoing development to a `development` branch and leave the `master` empty (maybe with a small README sending visitors to the `wazuh/wazuh` repository).

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+1. **Installing Dependencies on Debian**
+
+    To install the necessary dependencies on a Debian-based system, run the following commands:
+
+    ```bash
+    sudo apt-get update
+    sudo apt-get install git zip curl tar ninja-build pkg-config
+    ```
+
 1. **Clone the Repository**
```
Comment (TomasTurina) [inline BUILD.md:23]: Number are not OK, starting from this one.

File: BUILD.md [hunk 2]
```diff
@@ -39,14 +59,28 @@
     cmake .. -DBUILD_TESTS=1
     ```
 
-4. **Build the Project**
+1. **Build the Project**
```
Comment (TomasTurina) [inline BUILD.md:62]: This description is repeated.

File: BUILD.md (hunk 1 above)
Comment (jr0me) [reply BUILD.md:23]: 😅 it was on purpose. Markdown renders the number incrementally despite this. But I can change it of course.

Comment (jr0me) [reply BUILD.md]: Done

File: BUILD.md (hunk 2 above)
Comment (jr0me) [reply BUILD.md]: Fixed!

Comment (vikman90) [reply BUILD.md]: I think we can put it all together in one block:

```shell
cmake -B build
cmake --build build
```

The second command is more portable and guarantees that we're using the build options set by CMake. Normally, it's equivalent to `make`.

**Review by vikman90 — CHANGES_REQUESTED** [review]

File: BUILD.md (hunk 2 above)
Comment (jr0me) [reply BUILD.md]: Done!

**Review by TomasTurina — APPROVED** [review]

**Review by vikman90 — APPROVED** [review]

**Pull request merged**